code/obj/item/Bamboo_spike_trap.dm

```diff
@@ -0,0 +1,207 @@
+/obj/item/bamboo_spike_trap
+	name = "spike trap"
+	desc = "Curious is the trapmaker's art; his efficacy unwitnessed by his own eyes. Can be laced with chemicals."
+	icon = 'icons/obj/items/weapons.dmi'
+	icon_state = "Bamboo_Spike_Trap-Packet"
+	item_state = "table_parts"
+	flags = TABLEPASS | FPRINT | NOSPLASH
+	w_class = W_CLASS_NORMAL
+	force = 3
+	throwforce = 5
+	var/armed = FALSE ///This determinates if the trap is armed or not
+	var/armed_force = 9 ///how much damage the trap does when stepped upon
+	var/armed_weakened = 3 SECONDS ///how long you are weakened after stepping into the trap
+	var/crashed_force = 20 ///how much damage the trap does when crashed into
+	var/crashed_stun = 3 SECONDS ///how long you are stunned if you crash into the trap
+	var/reagent_storage = 15 ///How much the max amount of chems is the trap should be able to hold
+	var/transfer_multiplier = 0.5 ///Multiplier to damage to calculate the amount of chems tranferred
+	var/target_zone = "chest" ///which zone the trap tries to target and calculate the damage resist from
+	var/disarming_time = 2 SECONDS ///how long disarming with a wrench should take
+	var/arming_time = 2 SECONDS ///how long arming should take
+	var/break_down_time = 1 SECONDS ///how long an unarmed trap should take to disassemble with a wrench
+	var/break_down_amount = 3 ///how much bamboo is created when this gets broken down, should be less or equal to the amount needed to create it
+	stamina_damage = 0
+	stamina_cost = 0
+	stamina_crit_chance = 5
+
+	New()
+		..()
+		src.create_reagents(src.reagent_storage)
+
+	examine()
+		. = ..()
+		if (src.armed)
+			. += "<span class='alert'>It looks like it's armed.</span>"
+
+
+	attackby(obj/item/W, mob/user)
+		if(issnippingtool(W))
+			if (src.armed)
+				if (ON_COOLDOWN(user, "disarming_spike_trap", user.combat_click_delay))
+					return
+				playsound(src.loc, 'sound/items/Scissor.ogg', 60)
+				user.visible_message("[user] starts disarming [src]...")
+				var/datum/action/bar/icon/callback/action_bar = new /datum/action/bar/icon/callback(user, src, src.disarming_time,
+				/obj/item/bamboo_spike_trap/proc/disarm,\list(user), W.icon, W.icon_state, "[user] finishes disarming [src]")
+				actions.start(action_bar, user)
+				return
+			else
+				if (ON_COOLDOWN(user, "dismantling_spike_trap", user.combat_click_delay))
+					return
+				playsound(src.loc, 'sound/items/Scissor.ogg', 60)
+				user.visible_message("[user] starts breaking down [src]...")
+				var/datum/action/bar/icon/callback/action_bar = new /datum/action/bar/icon/callback(user, src, src.break_down_time,
+				/obj/item/bamboo_spike_trap/proc/break_down,\list(user), W.icon, W.icon_state, "[user] finishes breaking down [src]")
```

Comment:
this should be indented if you want to break it over two lines (same with other cases)